sys/js/timer.c

```diff
@@ -0,0 +1,89 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "event.h"
+#include "js.h"
+#include "xtimer.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+typedef struct {
+    js_callback_t callback;
+    jerry_value_t object;
+    xtimer_t xtimer;
+} js_xtimer_t;
+
+static void _js_xtimer_freecb(void *native_p)
+{
+    DEBUG("%s:l%u:%s()\n", __FILE__, __LINE__, __func__);
+
+    js_xtimer_t *js_xtimer = (js_xtimer_t *) native_p;
+    xtimer_remove(&js_xtimer->xtimer);
+    free(js_xtimer);
```

Comment:
@kaspar030 isn't jerryscript freeing this js_xtimer block?

Reply:
Well, jerryscript garbage-collects the javascript  timer object. If that happens to be a "native" object, it calls that native object's free_cb, which happens to be this function.

Reply:
yeah, I see. But why is it required to call `free()`? Shouldn't the js memory manager free the whole block?

Reply:
No, when the native timer object is created (using ```js_object_native_create()```), the js_xtimer_t is malloc'ed outside jerryscript's heap (so we still have control over it). Thus our binding code both allocateds and frees the object.

Reply:
ok, I see ;)